resolver/bootstrap_test.go

```diff
@@ -319,4 +398,33 @@ var _ = Describe("Bootstrap", Label("bootstrap"), func() {
 			})
 		})
 	})
+
+	Describe("multiple upstreams", func() {
+		var (
+			mockUpstream1 *MockUDPUpstreamServer
+			mockUpstream2 *MockUDPUpstreamServer
+		)
+
+		BeforeEach(func() {
+			mockUpstream1 = NewMockUDPUpstreamServer().WithAnswerRR("example.com 123 IN A 123.124.122.122")
+			DeferCleanup(mockUpstream1.Close)
+
+			mockUpstream2 = NewMockUDPUpstreamServer().WithAnswerRR("example.com 123 IN A 123.124.122.122")
+			DeferCleanup(mockUpstream1.Close)
+
+			sutConfig.BootstrapDNS = []config.BootstrappedUpstreamConfig{
+				{Upstream: mockUpstream1.Start()},
+				{Upstream: mockUpstream2.Start()},
+			}
+		})
+
+		It("uses both", func() {
+			_, err := sut.resolve("example.com.", []dns.Type{dns.Type(dns.TypeA)})
+
+			Expect(err).To(Succeed())
+
+			Expect(mockUpstream1.GetCallCount()).To(Equal(1))
+			Expect(mockUpstream2.GetCallCount()).To(Equal(1))
+		})
+	})
```

Comment:
This test was working locally, but on the CI it's flaky. I'll rewrite it differently.